docs/introduction-to-smart-contracts.rst

```diff
@@ -194,11 +194,10 @@ a "blockchain explorer" that tracks transactions and balances of your new coin.
 Blockchain Basics
 *****************
 
-Blockchains as a concept are not too hard to understand for programmers. The reason is that
-most of the complications (mining, `hashing <https://en.wikipedia.org/wiki/Cryptographic_hash_function>`_, `elliptic-curve cryptography <https://en.wikipedia.org/wiki/Elliptic_curve_cryptography>`_, `peer-to-peer networks <https://en.wikipedia.org/wiki/Peer-to-peer>`_, etc.)
-are just there to provide a certain set of features and promises. Once you accept these
-features as given, you do not have to worry about the underlying technology - or do you have
-to know how Amazon's AWS works internally in order to use it?
+Blockchains, as a concept, are not hard for programmers to understand. Much of the confusion
+comes from complicated features and promises (mining, `hashing <https://en.wikipedia.org/wiki/Cryptographic_hash_function>`_, `elliptic-curve cryptography <https://en.wikipedia.org/wiki/Elliptic_curve_cryptography>`_, `peer-to-peer networks <https://en.wikipedia.org/wiki/Peer-to-peer>`_, etc.),
```

Comment:
Hm, this sentence ("once they are understood, ...") sounds like you have to understand how e.g. elliptic curve cryptography works in order to use Ethereum, but this sentence should state exactly the opposite.

Reply:
Ok but in this case, you should not mention elliptic curve cryptography, because it suffices to understand public key cryptography. Peer-to-peer networks do not have to be understood at all.

Reply:
What I wanted to get across in the original version is that there are certain features and promises. For example the fact that the value of `msg.sender` cannot be faked. Those have to be accepted for granted, and it is irrelevant how they are provided and ensured (namely using mining, elliptic curve cryptography and so on).

Reply:
ah, okay, that makes things much more clear. Will update the wording around this to reflect that. Thanks!

Reply:
Wonderful, thanks!